src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -294,603 +363,1248 @@ public static boolean hasToolLevel(final String tool, final ItemStack stack, fin
+        return findFirstSlotInProviderWith(provider, getItemFromBlock(block), itemDamage);
     }
 
     /**
-     * Returns all <code>ItemStack</code>s in an inventory.
-     * Stores this in an array.
+     * Returns the index of the first occurrence of the Item with the given
+     * ItemDamage in the {@link ICapabilityProvider}.
      *
-     * @param inventory Inventory to return all item stacks from.
-     * @return Array of item stacks.
+     * @param provider   {@link ICapabilityProvider} to check
+     * @param targetItem Item to find.
+     * @param itemDamage The damage value to match on the stack.
+     * @return Index of the first occurrence
      */
-    @NotNull
-    public static ItemStack[] getAllItemStacks(@NotNull final IInventory inventory)
+    public static int findFirstSlotInProviderWith(@NotNull final ICapabilityProvider provider, final Item targetItem, int itemDamage)
     {
-        @NotNull final ItemStack[] itemStack = new ItemStack[inventory.getSizeInventory()];
-        for (int i = 0; i < inventory.getSizeInventory(); i++)
-        {
-            itemStack[i] = inventory.getStackInSlot(i);
-        }
-        return itemStack;
+        return findFirstSlotInProviderWith(provider, (ItemStack stack) -> compareItems(stack, targetItem, itemDamage));
     }
 
     /**
-     * Returns the amount of item stacks in an inventory.
-     * This equals {@link #getAllItemStacks(IInventory)}<code>.length();</code>.
+     * Returns the index of the first occurrence of an ItemStack that matches
+     * the given predicate in the {@link ICapabilityProvider}.
      *
-     * @param inventory Inventory to count item stacks of.
-     * @return Amount of item stacks in inventory.
+     * @param provider                    Provider to check
+     * @param itemStackSelectionPredicate The predicate to match.
+     * @return Index of the first occurrence
      */
-    public static int getAmountOfStacks(@NotNull final IInventory inventory)
+    public static int findFirstSlotInProviderWith(@NotNull final ICapabilityProvider provider, Predicate<ItemStack> itemStackSelectionPredicate)
     {
-        int count = 0;
-        for (int i = 0; i < inventory.getSizeInventory(); i++)
+        for (IItemHandler handler : getItemHandlersFromProvider(provider))
         {
-            final ItemStack is = inventory.getStackInSlot(i);
-
-            if (is != null && is != ItemStack.EMPTY)
+            int foundSlot = findFirstSlotInItemHandlerWith(handler, itemStackSelectionPredicate);
+            //TODO: When contract is hardened later: Replace this -1 check with a try-catch block.
+            if (foundSlot > -1)
             {
-                count++;
+                return foundSlot;
             }
         }
-        return count;
+
+        return -1;
+        //TODO: Later harden contract to remove compare on slot := -1
+        //throw new IllegalStateException("Item "+targetItem.getUnlocalizedName() + " not found in ItemHandler!");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)